src/mame/drivers/brother_lw350.cpp

```diff
@@ -0,0 +1,1920 @@
+	DECLARE_WRITE_LINE_MEMBER(fdc_dreq_w) { maincpu->set_input_line(Z180_INPUT_LINE_DREQ0, state ? ASSERT_LINE : CLEAR_LINE); }
+	DECLARE_READ_LINE_MEMBER(fdc_dend_r) { return maincpu->get_tend0(); }
+
+	// CRTC
+	MC6845_UPDATE_ROW(crtc_update_row);
+	MC6845_ON_UPDATE_ADDR_CHANGED(crtc_addr);
+	DECLARE_WRITE_LINE_MEMBER(crtc_vsync);
+	void io_72_w(uint8_t data) { io_72 = data; }
+	void io_73_w(uint8_t data) { io_73 = data; }
+	void io_74_w(uint8_t data);
+	void io_75_w(uint8_t data) { io_75 = data; }
+
+	uint8_t io_72, io_73, io_74, io_75; // gfx
+	uint8_t io_7a, io_b8, rombank;
+	uint32_t framecnt;
+
+	TIMER_DEVICE_CALLBACK_MEMBER(int1_timer_callback);
+
+protected:
+	// driver_device overrides
+	void machine_start() override;
+	void machine_reset() override;
+
+	void video_start() override;
+
+	uint8_t* rom{};
+	uint8_t* dict_rom{};
+	uint8_t framebuffer[720 * 350]; // pixel data
+	std::map<uint32_t, std::string> symbols;
+};
+
+void lw450_state::video_start()
+{
+}
+
+std::string lw450_state::pc()
+{
+	class z180_friend : z180_device { friend class lw450_state; };
+	auto cpu = static_cast<z180_friend*>(dynamic_cast<z180_device*>(&machine().scheduler().currently_executing()->device()));
+	offs_t phys = cpu->pc();
+	cpu->memory_translate(AS_PROGRAM, 0, phys);
+
+	return symbolize(phys);
+}
+
+std::string lw450_state::symbolize(uint32_t adr)
+{
+	if(symbols.empty())
+		return string_format("%06x", adr);
+
+	auto floor_it = symbols.lower_bound(adr);
+	if((floor_it == symbols.end() && !symbols.empty()) || floor_it->first != adr)
+		--floor_it;
+	if(floor_it != symbols.end())
+		return string_format("%s+%x (%06x)", floor_it->second, adr - floor_it->first, adr);
+	else
+		return string_format("%06x", adr);
+}
+
+std::string lw450_state::callstack()
+{
+	class z180_friend : z180_device { friend class lw450_state; };
+	auto cpu = static_cast<z180_friend*>(dynamic_cast<z180_device*>(&machine().scheduler().currently_executing()->device()));
+	offs_t pc = cpu->pc();
+	cpu->memory_translate(AS_PROGRAM, 0, pc);
+
+	//int depth = 0;
+	std::string output;
+	output += symbolize(pc) + " >> ";
+
+//	if(output.find("abort") != std::string::npos)
+//		__debugbreak();
+/*
+	// floppy routines
+	if(pc >= 0x73000 && pc <= 0x75000) {
+		offs_t sp = cpu->sp();
+		for(int i = 0; i < 4; i++) {
+			offs_t back = cpu->space(AS_PROGRAM).read_word_unaligned(sp);
+			cpu->memory_translate(AS_PROGRAM, 0, back);
+			if(back >= 0x73000 && back <= 0x75000) {
+				output += symbolize(back) + " >> ";
+				depth++;
+				if(depth < 8)
+					i = 0;
+			}
+			sp += 2;
+		}
+	}
+*/
+	return output;
+}
+
+void lw450_state::map_program(address_map& map)
+{
+	map(0x00000, 0x01fff).rom();
+	map(0x02000, 0x05fff).ram();
+	map(0x06000, 0x3ffff).rom();
+	map(0x40000, 0x5ffff).r(FUNC(lw450_state::rom40000_r)); // dictionary ROM, banked according to I/O E0
+	map(0x62000, 0x71fff).ram(); // D-RAM UPPER/LOWER
+	map(0x72000, 0x75fff).r(FUNC(lw450_state::rom72000_r)); // => ROM 0x02000-0x05fff
```

Comment:
Is it not possible to do this with `.rom` with an explicit region name and base?

Reply:
Probably, I don't know how to do it. Tips?

Reply:
done